src/components/autocomplete/autocomplete.spec.js

```diff
@@ -320,6 +320,111 @@ describe('<md-autocomplete>', function() {
       element.remove();
     }));
 
+    it('does not clear the value nor blur when hitting escape', inject(function($mdConstant, $timeout, $document) {
+      var scope = createScope();
+      var template = '\
+          <md-autocomplete\
+              md-escape-options="none"\
+              md-search-text="searchText"\
+              md-items="item in match(searchText)"\
+              md-item-text="item.display"\
+              placeholder="placeholder">\
+            <span md-highlight-text="searchText">{{item.display}}</span>\
+          </md-autocomplete>';
+      var element = compile(template, scope);
+      var ctrl = element.controller('mdAutocomplete');
+
+      // Focus the input
+      ctrl.focus();
+      $timeout.flush();
+
+      expect(scope.searchText).toBe('');
+
+      scope.$apply('searchText = "test"');
+
+      expect(scope.searchText).toBe('test');
+
+      $timeout.flush();
+      scope.$apply(function() {
+        ctrl.keydown(keydownEvent($mdConstant.KEY_CODE.ESCAPE));
+      });
+
+      expect(scope.searchText).toBe('test');
+      expect($document.activeElement).toBe(ctrl[0]);
+
+      element.remove();
+    }));
+
+    it('does not clear the value but does blur when hitting escape', inject(function($mdConstant, $timeout, $document) {
+      var scope = createScope();
+      var template = '\
+          <md-autocomplete\
+              md-escape-options="blur"\
+              md-search-text="searchText"\
+              md-items="item in match(searchText)"\
+              md-item-text="item.display"\
+              placeholder="placeholder">\
+            <span md-highlight-text="searchText">{{item.display}}</span>\
+          </md-autocomplete>';
+      var element = compile(template, scope);
+      var ctrl = element.controller('mdAutocomplete');
+
+      // Focus the input
+      ctrl.focus();
+      $timeout.flush();
+
+      expect(scope.searchText).toBe('');
+
+      scope.$apply('searchText = "test"');
+
+      expect(scope.searchText).toBe('test');
+
+      $timeout.flush();
+      scope.$apply(function() {
+        ctrl.keydown(keydownEvent($mdConstant.KEY_CODE.ESCAPE));
+      });
+
+      expect(scope.searchText).toBe('test');
+      expect($document.activeElement).toBe(undefined);
+
+      element.remove();
+    }));
+
+    it('clear the value but does not blur when hitting escape', inject(function($mdConstant, $timeout, $document) {
+      var scope = createScope();
+      var template = '\
+          <md-autocomplete\
+              md-escape-options="clear"\
+              md-search-text="searchText"\
+              md-items="item in match(searchText)"\
+              md-item-text="item.display"\
+              placeholder="placeholder">\
+            <span md-highlight-text="searchText">{{item.display}}</span>\
+          </md-autocomplete>';
```

Comment:
These templates seem pretty repetitive, can you try to do them in the before each or compile them in a function?